go/host/p2p/metrics.go

```diff
@@ -0,0 +1,60 @@
+package p2p
+
+import (
+	gethmetrics "github.com/ethereum/go-ethereum/metrics"
+	hostcommon "github.com/obscuronet/go-obscuro/go/common/host"
+)
+
+const (
+	_failedMessageRead        = "msg/inbound/failed_read"
+	_failedMessageDecode      = "msg/inbound/failed_decode"
+	_failedConnectSendMessage = "msg/outbound/failed_peer_connect"
+	_failedWriteSendMessage   = "msg/outbound/failed_write"
+	_receivedMessage          = "msg/inbound/success_received"
+)
+
+// metrics represents the metrics for the p2p library
+type metrics struct {
+	hostBasedGauges map[string]*perStringGaugeMap // gauges broken down per host
+}
+
+// newP2PMetrics creates the P2P metrics used by the P2P layer
+func newP2PMetrics(registry gethmetrics.Registry) *metrics {
+	return &metrics{
+		hostBasedGauges: map[string]*perStringGaugeMap{
+			_failedMessageRead:        newPerStringGaugeMap(registry, _failedMessageRead),
+			_failedMessageDecode:      newPerStringGaugeMap(registry, _failedMessageDecode),
+			_failedConnectSendMessage: newPerStringGaugeMap(registry, _failedConnectSendMessage),
+			_failedWriteSendMessage:   newPerStringGaugeMap(registry, _failedWriteSendMessage),
+			_receivedMessage:          newPerStringGaugeMap(registry, _receivedMessage),
+		},
+	}
+}
+
+// incrementHost adds one (1) to the given gauge instrument at a given host
+func (m *metrics) incrementHost(instrument string, host string) {
```

Comment:
a better name for this would be: "trackHostEvent".
Since you've absracted away the gauge, why leak that through the name?

Reply:
Good point. Changed the name to "trackHostEvent" and then when updating the description changed it again to "incrementHostEvent". I think it makes it clearer. What do you think ?